cmake/modules/AddSwift.cmake

```diff
@@ -119,6 +119,12 @@ function(_add_variant_c_compile_link_flags)
       "-B" "${SWIFT_ANDROID_PREBUILT_PATH}/arm-linux-androideabi/bin/")
   endif()
 
+  if("${CFLAGS_SDK}" STREQUAL "FUCHSIA")
+    list(APPEND result
+      "-I${SWIFT_FUCHSIA_BUILD_PATH}/buildtools/linux-x64/clang/lib/x86_64-fuchsia/include/c++/v1/"
+      "-L${SWIFT_FUCHSIA_BUILD_PATH}/buildtools/linux-x64/clang/lib/x86_64-fuchsia/lib")
+  endif()
```

Comment:
One big regret I have with the way I merged the Swift/Android CMake is the presence of hardcoded paths, such as `"${SWIFT_ANDROID_PREBUILT_PATH}/arm-linux-androideabi/lib/armv7-a"` or `"${SWIFT_ANDROID_PREBUILT_PATH}/lib/gcc/arm-linux-androideabi/${SWIFT_ANDROID_NDK_GCC_VERSION}.x"`. The problem is that the paths sometimes change, and so newer releases of the SDK/NDK break Swift's build.

Similarly, I think these would be better as externally settable values. If appropriate, the paths you use here could be provided as defaults, but I think allowing users to specify the path may prevent some build breakages later on.

Reply:
Yah previously I was debating between adding 8 different args to specify paths vs working off one path and baking in where things are in Fuchsia into the cmake files. I chose the latter originally because early on in the port I didn't know what paths swift would eventually need to finally compile. 

Right now I'm baking in:
* aarch64 sysroot path
* aarch64 userland shared lib path (for libicu*.so)
* x86_64 sysroot path
* x86_64 userland shared lib path (for libicu*.so)
* ICU's common include path 
* ICU's i18n include path
* fuchsia's toolchain libc++ include path (at different path between linux/darwin toolchains)
* fuchsia's toolchain lib path (at different path between linux/darwin toolchains)

All this I should clean up. I will likely need built-util args for each of these. I will also need an arg to specify which archs you want to build the stdlib for since your local build of fuchsia might only have built artifacts for one arch or you just want to make a leaner build locally.

Reply:
We really should consider upgrading the minimum CMake version for non-Darwin targets.  That way we could take advantage of the cross-compiling toolchain definitions from CMake which really simplify handling the android NDK.

Reply:
fixed.